gbdk-lib/examples/cross-platform/text_typewriter/src/main.c

```diff
@@ -0,0 +1,108 @@
+#include <gbdk/platform.h>
+#include <stdint.h>
+#include "Font.h"
+
+uint8_t GetCharacterVRamTile(char character) {
+
+    uint8_t vramTile=0;
+
+    // Char's can be interpreted as integers
+    // We don't need to map every alpha-numeric character
+    // We can use basic math to simplify A-Z and 0-9
+    if(character>='a'&&character<='z')vramTile = (character-'a')+1;
+    else if(character>='A'&&character<='Z')vramTile = (character-'A')+1;
+    else if(character>='0'&&character<='9')vramTile = (character-'0')+27;
+    else if(character=='!')vramTile = 37;
+    else if(character==':')vramTile = 38;
+    else if(character=='?')vramTile = 39;
+    else if(character=='/')vramTile = 40;
+    else if(character=='=')vramTile = 41;
+    else if(character==',')vramTile = 42;
+    else if(character=='.')vramTile = 43;
+    else if(character=='<')vramTile = 44;
+    else if(character=='>')vramTile = 45;
+
+    return vramTile;
+
+}
+
+
+void DrawText( char* text,uint8_t typewriterDelay){
+
+    uint8_t column = 0;
+    uint8_t row = 0;
+
+    // Get the address of the first tile in the row
+    uint8_t* vramAddress = get_bkg_xy_addr(column,row);
+
+    uint16_t index=0;
+    uint8_t columnIndex=0;
+
+    while(text[index]!='\0'){
+
+        char character = text[index];
+
+        // Get the vram tile to draw
+        uint8_t vramTile = GetCharacterVRamTile(character);
+
+        set_vram_byte(vramAddress++,vramTile);
+
+        #if defined(SEGA)
+        set_vram_byte(vramAddress++,0);
+        #endif
+
+        index++;
+        columnIndex++;
+
+        // if we've reached the end of the row
+        if(columnIndex>=DEVICE_SCREEN_WIDTH){
+            
+            // reset for the next row
+            vramAddress = get_bkg_xy_addr(column,++row);
+            
+            columnIndex=0;
+        }
+
+        if(typewriterDelay>0){
+
+            #if !defined(SEGA) && !defined(NINTENDO_NES)
```

Comment:
I'll leave the details to @untoxa and @michel-iwaniec since they're more familiar with the platforms, but it would be nice to have sound for SMS/GG and NES as well. it could be split out to a function to keep it manageable.

Reply:
Yes, of course. If you add sound to the cross-platform example, better make it work on all platforms. It it is not essential, then it is better to exclude that.

Reply:
I don't know how to do sound for those platforms. I also don't see any examples on it in their folders.

Reply:
There is some code in the port of Black Castle to GG/SMS and NES that I think attempts to vaguely emulate a GB sound interface for the other platforms.  @untoxa and @michel-iwaniec have a better understanding of the code there and whether it's appropriate to use. It may be that it's too special-cased to that game and they don't want users to take it as something to mimic.
https://github.com/untoxa/BlackCastle/blob/74dc83cb56df28182cdd0f2ae5019828e440f931/src/sfx.c

The alternative is to follow Toxa's suggestion and omit the sound feature since it's not essential.

Reply:
> There is some code in the port of Black Castle to GG/SMS and NES that I think attempts to vaguely emulate a GB sound interface for the other platforms. @untoxa and @michel-iwaniec have a better understanding of the code there and whether it's appropriate to use. It may be that it's too special-cased to that game and they don't want users to take it as something to mimic. https://github.com/untoxa/BlackCastle/blob/74dc83cb56df28182cdd0f2ae5019828e440f931/src/sfx.c
> 
> The alternative is to follow Toxa's suggestion and omit the sound feature since it's not essential.

audio removed from the typewriter and advanced dialogue examples